satpy/composites/__init__.py

```diff
@@ -945,79 +945,72 @@ def __call__(self, datasets, optional_datasets=None, **info):
             raise IncompatibleAreas('RatioSharpening requires datasets of '
                                     'the same size. Must resample first.')
 
-        new_attrs = {}
         optional_datasets = tuple() if optional_datasets is None else optional_datasets
         datasets = self.match_data_arrays(datasets + optional_datasets)
-        r = datasets[0]
-        g = datasets[1]
-        b = datasets[2]
+        red, green, blue, new_attrs = self._get_and_sharpen_rgb_data_arrays_and_meta(datasets, optional_datasets)
+        combined_info = self._combined_sharpened_info(info, new_attrs)
+        res = super(RatioSharpenedRGB, self).__call__((red, green, blue,), **combined_info)
+        res.attrs.pop("units", None)
+        return res
+
+    def _get_and_sharpen_rgb_data_arrays_and_meta(self, datasets, optional_datasets):
+        new_attrs = {}
+        red = datasets[0]
+        green = datasets[1]
+        blue = datasets[2]
         if optional_datasets and self.high_resolution_band is not None:
             LOG.debug("Sharpening image with high resolution {} band".format(self.high_resolution_band))
             high_res = datasets[3]
             if 'rows_per_scan' in high_res.attrs:
                 new_attrs.setdefault('rows_per_scan', high_res.attrs['rows_per_scan'])
             new_attrs.setdefault('resolution', high_res.attrs['resolution'])
             colors = ['red', 'green', 'blue']
-            low_res_idx = colors.index(self.high_resolution_band)
+            low_resolution_index = colors.index(self.high_resolution_band)
```

Comment:
I'm not sure I understand here: how can the position of the high resolution band name become the low resolution index?

oh, ok, got it now. To avoid confusion in he future, could the `red`, `green`, `blue` variables be renamed to `low_res_red`, `low_res_green`, `low_res_blue` and `colors` be renamed to `low_res_colors`?

Reply:
And maybe rename `self.high_resolution_band` to `self.high_resolution_band_name`?